bundles/automation/org.eclipse.smarthome.automation.event.test/src/test/groovy/org/eclipse/smarthome/automation/event/RuleEventTest.groovy

```diff
@@ -0,0 +1,179 @@
+        ]
+        def conditions = [
+            new Condition("ItemStateCondition3", "ItemStateCondition", condition1Config, null),
+            new Condition("ItemStateCondition4", "ItemStateCondition", condition2Config, null)
+        ]
+        def actions = [
+            new Action("ItemPostCommandAction2", "ItemPostCommandAction", actionConfig, null)
+        ]
+
+        def rule = new Rule("myRule21",triggers, conditions, actions, null, null)
+        rule.name="RuleEventTestingRule"
+
+        logger.info("Rule created: "+rule.getUID())
+
+        def ruleRegistry = getService(RuleRegistry)
+        ruleRegistry.add(rule)
+        ruleRegistry.setEnabled(rule.UID, true)
+
+        //TEST RULE
+
+        def EventPublisher eventPublisher = getService(EventPublisher)
+        def ItemRegistry itemRegistry = getService(ItemRegistry)
+        SwitchItem myMotionItem = itemRegistry.getItem("myMotionItem2")
+        Command commandObj = TypeParser.parseCommand(myMotionItem.getAcceptedCommandTypes(), "ON")
+        eventPublisher.post(ItemEventFactory.createCommandEvent("myPresenceItem2", commandObj))
+
+        Event itemEvent = null
+
+        def itemEventHandler = [
+            receive: {  Event e ->
+                logger.info("Event: " + e.topic)
+                if (e.topic.contains("myLampItem2")){
+                    itemEvent=e
+                }
+            },
+
+            getSubscribedEventTypes: {
+                Sets.newHashSet(ItemUpdatedEvent.TYPE, ItemStateEvent.TYPE)
+            },
+
+            getEventFilter:{ null }
+
+        ] as EventSubscriber
+
+        registerService(itemEventHandler)
+        commandObj = TypeParser.parseCommand(itemRegistry.getItem("myMotionItem2").getAcceptedCommandTypes(),"ON")
+        eventPublisher.post(ItemEventFactory.createCommandEvent("myMotionItem2", commandObj))
+        waitForAssert ({ assertThat itemEvent, is(notNullValue())} , 3000, 100)
+        assertThat itemEvent.topic, is(equalTo("smarthome/items/myLampItem2/state"))
+        assertThat (((ItemStateEvent)itemEvent).itemState, is(OnOffType.ON))
+        def myLampItem2 = itemRegistry.getItem("myLampItem2")
+        assertThat myLampItem2, is(notNullValue())
+        logger.info("myLampItem2 State: " + myLampItem2.state)
+        assertThat myLampItem2.state, is(OnOffType.ON)
+        assertThat ruleEvents.size(), is(not(0))
+        assertThat ruleEvents.find{it.topic =="smarthome/rules/myRule21/added"}, is(notNullValue())
+        assertThat ruleEvents.find{it.topic =="smarthome/rules/myRule21/state"}, is(notNullValue())
+        def stateEvents = ruleEvents.findAll{it.topic=="smarthome/rules/myRule21/state"} as List<RuleStatusInfoEvent>
+        assertThat stateEvents, is(notNullValue())
+        def runningEvent = stateEvents.find{it.statusInfo.status==RuleStatus.RUNNING}
+        assertThat runningEvent, is(notNullValue())
+        ruleRegistry.remove("myRule21")
```

Comment:
The test will fail here because Condition and CondidionImpl, both declare a field connections. The Json serializer won't make it.

This issue belongs to DTO structure!